code/modules/mob/living/brain/posibrain.dm

```diff
@@ -196,3 +242,46 @@ GLOBAL_VAR(posibrain_notify_cooldown)
 		icon_state = "[initial(icon_state)]-occupied"
 	else
 		icon_state = initial(icon_state)
+
+//This Proc triggers when the Z level changes. If the Posi enters the station level, add it to the Job list.
+//If it leaves, remove it.
+/obj/item/mmi/posibrain/onTransitZ(old_z, new_z)
+	.=..()
+
+	if( is_station_level(old_z) == is_station_level(new_z))
+		//Early Return if we aren't entering or leaving station Z level.
+		return
+
+	if(is_occupied())
+		//No need to track occupied Posis
+		return
+
+	//Posi was on station, now is not on station
+	if( is_station_level(old_z) && !is_station_level(new_z))
+		GLOB.on_station_posis -= src
+
+	//Posi was off station, now is on station
+	if( !is_station_level(old_z) && is_station_level(new_z))
+		GLOB.on_station_posis += src
+
+	//Update Job Quantities
+	var/datum/job/job = SSjob.GetJob(JOB_NAME_POSIBRAIN)
+
+	//We should never show a posibrain as a filled job, so just make number of current positions zero
+	job.current_positions = 0
+
+	job.total_positions = length(GLOB.on_station_posis)
+
+/obj/item/mmi/posibrain/Destroy()
+
+	if(is_occupied())
+		//No need to track occupied Posis
+		return ..()
+
+	//If Posi is on station, remove it from the list.
+	if( src in GLOB.on_station_posis)
+		GLOB.on_station_posis -= src
+		var/datum/job/job = SSjob.GetJob(JOB_NAME_POSIBRAIN)
+		job.total_positions = length(GLOB.on_station_posis)
```

Comment:
the codes in `onTransitZ()` checking positron z and manipulating glob list can be merged in a proc.